l10n_it_dichiarazione_intento/models/account_invoice.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Francesco Apruzzese <f.apruzzese@apuliasoftware.it>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    dichiarazione_intento_ids = fields.Many2many(
+        'dichiarazione.intento', string='Dichiarazioni di Intento', copy=False)
+    exclude_from_dichiarazione_intento = fields.Boolean(
+        "Escludi da dich. d'intento")
+
+    @api.multi
+    def _set_fiscal_position(self):
+        for invoice in self:
+            if invoice.partner_id and invoice.date_invoice and invoice.type:
+                dichiarazioni = self.env['dichiarazione.intento'].get_valid(
+                    invoice.type.split('_')[0],
+                    invoice.partner_id.id,
+                    invoice.date_invoice)
+                if dichiarazioni:
+                    invoice.fiscal_position_id = \
+                        dichiarazioni.fiscal_position_id.id
```

Comment:
E' corretto che si possano creare più di una dichiarazione d'intento per Tipo - Partner - Intervallo di date ?

In questo caso c'è un errore : 
  File "/usr/local/lib/python3.5/dist-packages/odoo/addons/l10n_it_dichiarazione_intento/models/account_invoice.py", line 28, in _set_fiscal_position
    dichiarazioni.fiscal_position_id.id
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 942, in __get__
    record.ensure_one()
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 4393, in ensure_one
    raise ValueError("Expected singleton: %s" % self)

Reply:
Potrebbe essere mi sa, visto che ci potrebbero essere diverse vendite con diverse dichiarazioni. (curiosità: stai usando la 10.0 su python 3.5?)

Reply:
> (curiosità: stai usando la 10.0 su python 3.5?)
no, stavo migrando il modulo sulla 11, ma l'errore è lo stesso sulla 10